<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ flashfm</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">flashfm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/flashfm.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jennasimit/flashfm">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="FLASHFMwithJAM.html">FLASHFMwithJAM()</a></code> </p>
        </td>
        <td><p>Wrapper to run single-trait fine-mapping with JAMexpandedCor.multi on each trait, followed by flashfm and then constuct SNP groups for each approach and summarises results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="FLASHFMwithJAMhat.html">FLASHFMwithJAMhat()</a></code> </p>
        </td>
        <td><p>Wrapper to run single-trait fine-mapping with JAMexpandedCor.multi on each trait, followed by flashfm (using fast approximation version) and then constuct SNP groups for each approach and summarises results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="FMconfig.html">FMconfig</a></code> </p>
        </td>
        <td><p>List of FINEMAP config file contents for two simulated traits</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="FMtables.html">FMtables()</a></code> </p>
        </td>
        <td><p>Formats flashfm output to table format</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="JAM_PointEstimates.html">JAM_PointEstimates()</a></code> </p>
        </td>
        <td><p>internal function for multibeta, modified from JAM_PointEstimates_Package_Simplified of R2BGLiMS package (Paul Newcombe)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="JAM_PointEstimates_Xcov.html">JAM_PointEstimates_Xcov()</a></code> </p>
        </td>
        <td><p>internal function for multibeta, modified from JAM_PointEstimates_Package_Simplified of R2BGLiMS package</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="JAMexpanded.multi.html">JAMexpanded.multi()</a></code> </p>
        </td>
        <td><p>Expanded version of JAM (a single-trait fine-mapping approach) that first runs on thinned SNPs and then expands models on tag SNPs; this can run independently on multiple traits</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="JAMexpandedCor.multi.html">JAMexpandedCor.multi()</a></code> </p>
        </td>
        <td><p>Expanded version of JAM (a single-trait fine-mapping approach) that first runs on thinned SNPs and then expands models on tag SNPs; this can run independently on multiple traits</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="JAMmain.input.html">JAMmain.input</a></code> </p>
        </td>
        <td><p>Output from JAMexpanded.multi for two simulated traits</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="N.html">N</a></code> </p>
        </td>
        <td><p>Vector of samples sizes for two simulated traits</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="Neff.html">Neff()</a></code> </p>
        </td>
        <td><p>Approximate effective sample size from GWAS of related samples</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="PPsummarise.html">PPsummarise()</a></code> </p>
        </td>
        <td><p>Summarise PP and MPP results from single-trait fine-mapping and flashfm, by SNP and by SNP group</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="Sxy.hat.html">Sxy.hat()</a></code> </p>
        </td>
        <td><p>estimates cross-product of each SNP with one trait</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="Vres.all.html">Vres.all()</a></code> </p>
        </td>
        <td><p>variance of model residuals for trait T1 at all models that have joint effect estimates</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="Vres.hat.html">Vres.hat()</a></code> </p>
        </td>
        <td><p>variance of model residuals for trait T1 at model index imod</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="Vx.hat.html">Vx.hat()</a></code> </p>
        </td>
        <td><p>Internal function, Vx.hat</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="X.html">X</a></code> </p>
        </td>
        <td><p>Simulated genotype matrix used to generate two traits</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="accessors.html">snps()</a></code> <code><a href="accessors.html">tags()</a></code> </p>
        </td>
        <td><p>Accessors for groups objects</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="allC12.html">allC12()</a></code> </p>
        </td>
        <td><p>internal function for calcAdjPP for that gives list of covariance matrix of residuals for all trait pairs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="best.models.cpp.html">best.models.cpp()</a></code> </p>
        </td>
        <td><p>Best models from a snpmpd object by cpp or maximum number of models - modification of best.models from GUESSFM by Chris Wallace</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="best.snps.html">best.snps()</a></code> </p>
        </td>
        <td><p>Best SNPs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="beta.html">beta</a></code> </p>
        </td>
        <td><p>List of vectors of single-SNP effects (GWAS) for two simulated traits</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calc.maxmin.html">calc.maxmin()</a></code> </p>
        </td>
        <td><p>calculate max or min of subset of a matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calcABF.html">calcABF()</a></code> </p>
        </td>
        <td><p>Calculate approximate Bayes' factor (ABF)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calcAdjPP.html">calcAdjPP()</a></code> </p>
        </td>
        <td><p>Calculates trait-adjusted posterior probabilities for all traits at sharing parameter kappa</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calcCres12.html">calcCres12()</a></code> </p>
        </td>
        <td><p>covariance between residuals of a pair of models for  a trait pair</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calcD12.html">calcD12()</a></code> </p>
        </td>
        <td><p>internal function for calcAdjPP for the 2-trait case</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calcDcon.html">calcDcon()</a></code> </p>
        </td>
        <td><p>internal function for calcAdjPP that gives constant term for delta</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calcQ12.html">calcQ12()</a></code> </p>
        </td>
        <td><p>internal function for calcAdjPP for a pair of traits</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="conversion.html">convert()</a></code> </p>
        </td>
        <td><p>Convert from old definitions of groups, tags classes to new</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cor.refdata.fn.html">cor.refdata.fn()</a></code> </p>
        </td>
        <td><p>Thin genotype correlation matrix for JAM input</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cor2cov.html">cor2cov()</a></code> </p>
        </td>
        <td><p>Correlation Matrix to Covariance Matrix Conversion</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="covY.html">covY</a></code> </p>
        </td>
        <td><p>Covariance matrix for two simulated traits</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cvs.html">cvs</a></code> </p>
        </td>
        <td><p>Vector of rsids for causal variants of two simulated traits</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="expand.mod.html">expand.mod()</a></code> </p>
        </td>
        <td><p>internal function for expanding models by tag SNPs in JAMexpanded.multi</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="flashfm.html">flashfm()</a></code> </p>
        </td>
        <td><p>Marginal PP for models of a set of traits, sharing information between the traits</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="flashfm.input.html">flashfm.input()</a></code> </p>
        </td>
        <td><p>Key input for flashfm - constructs snpmod object list and joint effect estimates list for all trait if have external single trait fine-mapping results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fm.html">fm</a></code> </p>
        </td>
        <td><p>Output from running FLASHFMwithJAM on the simulated dataset in this package</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="groupIDs.fn.html">groupIDs.fn()</a></code> </p>
        </td>
        <td><p>Find SNP group ids for a set of SNPs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="groupmulti.html">groupmulti()</a></code> </p>
        </td>
        <td><p>Group SNPs; adapted from group.multi by Chris Wallace</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="groups-class.html">groups-class</a></code> </p>
        </td>
        <td><p>Group focused class for holding information about sets of SNPs
defined by their mutual LD</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="groups-subset.html">tagsof()</a></code> <code><a href="groups-subset.html">taggedby()</a></code> <code><a href="groups-subset.html">`[`(<i>&lt;groups&gt;</i>,<i>&lt;character&gt;</i>,<i>&lt;missing&gt;</i>,<i>&lt;missing&gt;</i>)</a></code> <code><a href="groups-subset.html">`[`(<i>&lt;groups&gt;</i>,<i>&lt;numeric&gt;</i>,<i>&lt;missing&gt;</i>,<i>&lt;missing&gt;</i>)</a></code> <code><a href="groups-subset.html">`[`(<i>&lt;groups&gt;</i>,<i>&lt;logical&gt;</i>,<i>&lt;missing&gt;</i>,<i>&lt;missing&gt;</i>)</a></code> <code><a href="groups-subset.html">`[`(<i>&lt;tags&gt;</i>,<i>&lt;character&gt;</i>,<i>&lt;missing&gt;</i>,<i>&lt;missing&gt;</i>)</a></code> <code><a href="groups-subset.html">`[[`(<i>&lt;groups&gt;</i>,<i>&lt;numeric&gt;</i>)</a></code> <code><a href="groups-subset.html">`[[`(<i>&lt;groups&gt;</i>,<i>&lt;logical&gt;</i>)</a></code> <code><a href="groups-subset.html">`[[`(<i>&lt;groups&gt;</i>,<i>&lt;character&gt;</i>)</a></code> </p>
        </td>
        <td><p><code>tagsof</code> shows tags for a named character vector of SNPs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="logsum.html">logsum()</a></code> </p>
        </td>
        <td><p>logsum</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="makeNlist.html">makeNlist()</a></code> </p>
        </td>
        <td><p>Sample size information needed for flashfm</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="makeNlist.rel.html">makeNlist.rel()</a></code> </p>
        </td>
        <td><p>Sample size information needed for flashfm, when samples are related and have effective sample sizes for each trait</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="makeSNPgroups.html">makeSNPgroups()</a></code> </p>
        </td>
        <td><p>Make SNP groups using fine-mapping information from all of the traits</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="makeSNPgroups2.html">makeSNPgroups2()</a></code> </p>
        </td>
        <td><p>Make two sets of SNP groups using fine-mapping information from all of the traits using two sets of results and maps the names between them</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="makemod.html">makemod()</a></code> </p>
        </td>
        <td><p>internal function makemod</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="makeplink.html">makeplink()</a></code> </p>
        </td>
        <td><p>Convert genotype score data to plink ped-map format</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="marg.snps.html">marg.snps()</a></code> </p>
        </td>
        <td><p>internal function marg.snps</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="marg.snps.vecs.html">marg.snps.vecs()</a></code> </p>
        </td>
        <td><p>internal function marg.snps.vec</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="marginalpp.html">marginalpp()</a></code> </p>
        </td>
        <td><p>Marginal PP for models sharing information between traits</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mod.fn.html">mod.fn()</a></code> </p>
        </td>
        <td><p>internal processing function for JAMexpanded.multi</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="multibeta.html">multibeta()</a></code> </p>
        </td>
        <td><p>Using summary statistics, calculates joint effect estimates</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="overlap.html">overlap()</a></code> </p>
        </td>
        <td><p>overlap</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ppnsnp-class.html">ppnsnp-class</a></code> </p>
        </td>
        <td><p>Class to hold results of pp.nsnp</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="prune_maximal.html">prune_maximal()</a></code> </p>
        </td>
        <td><p>internal function for ref.data.fn</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="raf.html">raf</a></code> </p>
        </td>
        <td><p>Vector of RAFs from the simulated genotype matrix X used to generate two traits</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="refdata.fn.html">refdata.fn()</a></code> </p>
        </td>
        <td><p>Thin genotype matrix for JAM input</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="score2alleles.html">score2alleles()</a></code> </p>
        </td>
        <td><p>internal function for makeplink</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show-methods.html">show(<i>&lt;snpmod&gt;</i>)</a></code> <code><a href="show-methods.html">show(<i>&lt;snppicker&gt;</i>)</a></code> <code><a href="show-methods.html">show(<i>&lt;ppnsnp&gt;</i>)</a></code> <code><a href="show-methods.html">show(<i>&lt;tags&gt;</i>)</a></code> <code><a href="show-methods.html">show(<i>&lt;groups&gt;</i>)</a></code> </p>
        </td>
        <td><p>Show</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="snpin.html">snpin()</a></code> </p>
        </td>
        <td><p>Check whether a snp is in a snppicker, groups or tags object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="snpinfo.html">snpinfo</a></code> </p>
        </td>
        <td><p>Details of SNPs from the simulated genotype matrix X used to generate two traits</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="snpmod-class.html">snpmod-class</a></code> </p>
        </td>
        <td><p>Class to hold data relating to multiple models fitted to SNP data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="snppicker-class.html">snppicker-class</a></code> </p>
        </td>
        <td><p>Class to hold results of snp.picker algorithm</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="summary.html">summary(<i>&lt;snppicker&gt;</i>)</a></code> <code><a href="summary.html">summary(<i>&lt;groups&gt;</i>)</a></code> </p>
        </td>
        <td><p>Summaries</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="summaryStats.html">summaryStats()</a></code> </p>
        </td>
        <td><p>Summary statistics needed for flashfm input</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="tag.html">tag()</a></code> </p>
        </td>
        <td><p>tag</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="tagexpand.mod.html">tagexpand.mod()</a></code> </p>
        </td>
        <td><p>internal function for expanding models by tag SNPs in JAMexpandedCor.multi</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="tags-class.html">tags-class</a></code> </p>
        </td>
        <td><p>Tags focused class for holding information about sets of SNPs
defined by their mutual LD</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="union.html">union()</a></code> </p>
        </td>
        <td><p>Create a union of groups, snppicker or tags objects</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ybar.html">ybar</a></code> </p>
        </td>
        <td><p>Mean vector for two simulated traits</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#section-all-functions">All functions</a></li>
    </ul>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Jenn Asimit, Nico Hernandez, Chris Wallace.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


