<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Expanded version of JAM (a single-trait fine-mapping approach) that first runs on thinned SNPs and then expands models on tag SNPs; this can run independently on multiple traits — JAMexpanded.multi • flashfm</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Expanded version of JAM (a single-trait fine-mapping approach) that first runs on thinned SNPs and then expands models on tag SNPs; this can run independently on multiple traits — JAMexpanded.multi" />
<meta property="og:description" content="Expanded version of JAM (a single-trait fine-mapping approach) that first runs on thinned SNPs and then expands models on tag SNPs; this can run independently on multiple traits" />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">flashfm</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/flashfm.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Expanded version of JAM (a single-trait fine-mapping approach) that first runs on thinned SNPs and then expands models on tag SNPs; this can run independently on multiple traits</h1>
    
    <div class="hidden name"><code>JAMexpanded.multi.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Expanded version of JAM (a single-trait fine-mapping approach) that first runs on thinned SNPs and then expands models on tag SNPs; this can run independently on multiple traits</p>
    </div>

    <pre class="usage"><span class='fu'>JAMexpanded.multi</span>(
  <span class='no'>beta1</span>,
  <span class='no'>Gmat</span>,
  <span class='no'>snpinfo</span>,
  <span class='no'>ybar</span>,
  <span class='no'>Vy</span>,
  <span class='no'>N</span>,
  <span class='kw'>chr</span> <span class='kw'>=</span> <span class='fl'>10</span>,
  <span class='no'>fstub</span>,
  <span class='kw'>mafthr</span> <span class='kw'>=</span> <span class='fl'>0.005</span>,
  <span class='no'>path2plink</span>,
  <span class='kw'>r2</span> <span class='kw'>=</span> <span class='fl'>0.99</span>,
  <span class='no'>save.path</span>,
  <span class='kw'>related</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>y</span> <span class='kw'>=</span> <span class='kw'>NULL</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>beta1</th>
      <td><p>list where each component is a named vector</p></td>
    </tr>
    <tr>
      <th>Gmat</th>
      <td><p>genotype matrix (reference or from sample) where SNPs are columns, indivudals are rows</p></td>
    </tr>
    <tr>
      <th>snpinfo</th>
      <td><p>a data.frame with columns: snpid, BP, allele1, allele2; BP is base pair position, allele1 is reference allele</p></td>
    </tr>
    <tr>
      <th>ybar</th>
      <td><p>vector of trait means; if related samples, this should be based on unrelated samples; if traits are transformed to be standard Normal, could set ybar as 0-vector</p></td>
    </tr>
    <tr>
      <th>Vy</th>
      <td><p>vector of trait variances; if related samples, this should be based on unrelated samples; if traits are transformed to be standard Normal, could set Vy as 1-vector</p></td>
    </tr>
    <tr>
      <th>N</th>
      <td><p>vector of sample sizes for each trait; if related samples then give effective sample sizes</p></td>
    </tr>
    <tr>
      <th>chr</th>
      <td><p>chromosome number</p></td>
    </tr>
    <tr>
      <th>fstub</th>
      <td><p>file stub for path with file prefix to save intermediate files (plink format and tag snp list)</p></td>
    </tr>
    <tr>
      <th>mafthr</th>
      <td><p>MAF threshold for filtering SNPs to include in fine-mapping; default is 0, assuming pre-filterec</p></td>
    </tr>
    <tr>
      <th>path2plink</th>
      <td><p>file path to plink</p></td>
    </tr>
    <tr>
      <th>r2</th>
      <td><p>r.squared threshold for thinning SNPs before JAM</p></td>
    </tr>
    <tr>
      <th>save.path</th>
      <td><p>path to save JAM output files; tmp files and could delete these later</p></td>
    </tr>
    <tr>
      <th>related</th>
      <td><p>logical indicating if samples are related (TRUE) or not (FALSE); default is FALSE</p></td>
    </tr>
    <tr>
      <th>y</th>
      <td><p>if available, matrix of trait measurements or indicators of non-NA trait measurements (columns are traits); 
used to get joint sample counts; default is NULL and if not provided an approximation is used based on vector of trait sample sizes</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>list with 3 components: SM a list of snpmod objects giving fine-mapping results for each trait; mbeta a list of joint effect estimates for each trait; nsnps number of SNPs</p>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
    </ul>

    <h2>Author</h2>
    <p>Jenn Asimit</p>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Jenn Asimit, Nico Hernandez, Chris Wallace.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


